src/main/java/edu/bu/metcs673/trackr/user/TrackrUserServiceImpl.java

```diff
@@ -97,12 +97,12 @@ public String createUser(TrackrUserDTO userInput) {
     @Override
     public TrackrUserDTO updateUser(TrackrUserDTO dto) {
         TrackrUser trackrUser = getCurrentUser();
-        regexPasswordValidation(dto.getPassword());
 
         if (StringUtils.isNotBlank(dto.getNewPassword())) {
             if (!bCryptPasswordEncoder.matches(dto.getPassword(), trackrUser.getPassword())) {
                 throw new TrackrInputValidationException(CommonConstants.WRONG_PASSWORD);
             } else {
+                regexPasswordValidation(dto.getPassword());
```

Comment:
You need validate the new password in the change password workflow not the old one.

Reply:
I have modified this part but I can't test it now since my page is missing the change password button